src/nvim/map.c

```diff
@@ -123,29 +123,8 @@
     kh_clear(T##_##U##_map, map->table); \
   }
 
-static inline khint_t String_hash(String s)
```

Comment:
I am wondering whether `String_hash` stopping at NUL byte, so that `abc\0def` and `abc\0ghi` have the same hash is intentional?